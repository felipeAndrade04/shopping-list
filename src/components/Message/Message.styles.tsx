import styled from 'styled-components/native';

export const Container = styled.View`
  justify-content: center;
  width: 100%;
`;

export const Image = styled.Image`
  width: 240px;
  height: 200px;
  margin: 0 auto;
  margin-bottom: 50px;
`;

export const Title = styled.Text`
  font-size: 24px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.dark};
`;

export const Description = styled.Text`
  font-size: 16px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.dark};
  margin-top: 14px;
`;
