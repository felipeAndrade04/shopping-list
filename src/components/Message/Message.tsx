import React from 'react';
import * as S from './Message.styles';
import emptyAsset from '../../assets/images/empty.png';
import { MessageProps } from '.';

export function Message({ description, title }: MessageProps) {
  return (
    <S.Container>
      <S.Image source={emptyAsset} />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
    </S.Container>
  );
}
