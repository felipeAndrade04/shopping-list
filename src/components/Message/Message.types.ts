export interface MessageProps {
  title: string;
  description: string;
  showImage?: boolean;
}
