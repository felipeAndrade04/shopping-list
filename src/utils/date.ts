const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
