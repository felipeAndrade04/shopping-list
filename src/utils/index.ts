export * from './delay';
export * from './date';
