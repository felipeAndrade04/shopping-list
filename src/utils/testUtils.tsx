import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
