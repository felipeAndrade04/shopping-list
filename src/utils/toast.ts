import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
