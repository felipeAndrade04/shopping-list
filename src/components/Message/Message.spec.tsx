import { render } from '@testing-library/react-native';
import { Message } from './Message';
import { wrapper } from '@app/utils/testUtils';

const message = {
  title: 'Title teste',
  description: 'Description teste',
};

describe('<Message />', () => {
  it('Should be render correctly', () => {
    const tree = render(<Message title={message.title} description={message.description} />, {
      wrapper,
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should be render with image', () => {
    const { getByTestId, getByText } = render(
      <Message title={message.title} description={message.description} />,
      { wrapper }
    );

    expect(getByText(message.title)).toBeTruthy();
    expect(getByText(message.description)).toBeTruthy();
    expect(getByTestId('message-image')).toBeTruthy();
  });

  it('Should be render without image', () => {
    const { queryByTestId, getByText } = render(
      <Message showImage={false} title={message.title} description={message.description} />,
      { wrapper }
    );

    expect(getByText(message.title)).toBeTruthy();
    expect(getByText(message.description)).toBeTruthy();
    expect(queryByTestId('message-image')).toBeFalsy();
  });
});
