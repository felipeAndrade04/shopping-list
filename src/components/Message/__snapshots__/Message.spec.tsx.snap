// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Should be render correctly 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      [
        {
          "height": 200,
          "marginBottom": 50,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 0,
          "width": 240,
        },
      ]
    }
    testID="message-image"
  />
  <Text
    style={
      [
        {
          "color": "#0D0A0B",
          "fontFamily": "Poppins-Medium",
          "fontSize": 24,
        },
      ]
    }
  >
    Title teste
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0A0B",
          "fontFamily": "Poppins-Regular",
          "fontSize": 16,
          "marginTop": 14,
        },
      ]
    }
  >
    Description teste
  </Text>
</View>
`;
